clients/client-python/gravitino/client/generic_model_catalog.py

```diff
@@ -0,0 +1,534 @@
+        """
+        self._check_model_ident(model_ident)
+
+        model_full_ns = self._model_full_namespace(model_ident.namespace())
+        resp = self.rest_client.delete(
+            f"{self._format_model_request_path(model_full_ns)}/{encode_string(model_ident.name())}",
+            error_handler=MODEL_ERROR_HANDLER,
+        )
+        drop_resp = DropResponse.from_json(resp.body, infer_missing=True)
+        drop_resp.validate()
+
+        return drop_resp.dropped()
+
+    def list_model_versions(self, model_ident: NameIdentifier) -> List[int]:
+        """List all the versions of the register model by NameIdentifier in the catalog.
+
+        Args:
+            model_ident: The identifier of the model.
+
+        Raises:
+            NoSuchModelException: If the model does not exist.
+
+        Returns:
+            A list of model versions.
+        """
+        self._check_model_ident(model_ident)
+
+        model_full_ident = self._model_full_identifier(model_ident)
+        resp = self.rest_client.get(
+            self._format_model_version_request_path(model_full_ident),
+            error_handler=MODEL_ERROR_HANDLER,
+        )
+        model_version_list_resp = ModelVersionListResponse.from_json(
+            resp.body, infer_missing=True
+        )
+        model_version_list_resp.validate()
+
+        return model_version_list_resp.versions()
+
+    def get_model_version(
+        self, model_ident: NameIdentifier, version: int
+    ) -> ModelVersion:
+        """Get a model version by its identifier and version.
+
+        Args:
+            model_ident: The identifier of the model.
+            version: The version of the model.
+
+        Raises:
+            NoSuchModelVersionException: If the model version does not exist.
+
+        Returns:
+            The model version object.
+        """
+        self._check_model_ident(model_ident)
+
+        model_full_ident = self._model_full_identifier(model_ident)
+        resp = self.rest_client.get(
+            f"{self._format_model_version_request_path(model_full_ident)}/versions/{version}",
+            error_handler=MODEL_ERROR_HANDLER,
+        )
+        model_version_resp = ModelVersionResponse.from_json(
+            resp.body, infer_missing=True
+        )
+        model_version_resp.validate()
+
+        return GenericModelVersion(model_version_resp.model_version())
+
+    def model_version_exists(self, model_ident: NameIdentifier, version: int) -> bool:
+        """Check if the model version exists in the catalog.
+
+        Args:
+            model_ident: The identifier of the model.
+            version: The version of the model.
+
+        Returns:
+            True if the model version exists, false otherwise.
+        """
+        try:
+            self.get_model_version(model_ident, version)
+            return True
+        except NoSuchModelVersionException:
+            return False
+
+    def get_model_version_by_alias(
```

Comment:
My feeling is that the name may be misleading, users may feel that this method is to get the alias, not the model version.

Reply:
I was comparing this to the `get_model_version` method, which can be interpreted as "getting a model by its specific version".
If that is true, we can name this to `get_model_alias`.
After revisiting this collection of methods, I see what this method was designed for.

Reply:
Let me clarify the concept of model management in Gravitino. "Model" is a concept mapping to an ML model, "Model Version" is a snapshot of the specific "Model", one "Model" can have multiple "Model Versions". Besides, one "Model Version" can have multiple aliases, so users can use either "version" or "alias" to get a specific "Model Version".

So here, `get_model_version` means get a specific model version entity by version number, and `get_model_version` means get a specific model version entity by alias. It is not getting a model, it is getting a model version.

Reply:
Okay. Now I think I get the intention behind the "model version" thing and I know where all these come from.

TBH, the concept of "model version" as an entity was not so straightforward to me,
because "version" in my mind was not supposed to be treated as first-class entities.
A "version" has to be associated with a model, it is not supposed to managed separately. The "version" value can be associated with more than one model,
if we are using this term to represent a [semantic version](https://semver.org/).
I did get that the "version" in this context has nothing to do with semantic 
version, it is actually a sequence number, used to form a unique key for an entity.

Now we have a dilemma. We still need an attribute for "version" in the traditional
sense. So we invented a new term "alias", which further complicates the conceptual
design as a whole. Can/shall we promote "alias" to be a version string?

Reply:
Hi @tengqm , the term "version" and "alias" are common concepts in model management, it's not us alone define this word, you can also check here (https://docs.wandb.ai/guides/model_registry/model-management-concepts/) and here (https://docs.databricks.com/en/machine-learning/manage-model-lifecycle/index.html).